### PR TITLE
fix(cve): Search by CVE title/id (full or partial) when listing an im…

### PR DESCRIPTION
**Pull request by aokirisaki**
…age's CVEs

**What type of PR is this?**

**Which issue does this PR fix**:
#1242 

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/extensions/search/schema.graphql [hunk 1]
```diff
@@ -550,6 +550,8 @@ type Query {
         image: String!,
         "Sets the parameters of the requested page"
         requestedPage: PageInput
+        "Search term for specific CVE by title/id"
+        searchedCVE: String!
```
Comment (andaaron) [inline pkg/extensions/search/schema.graphql:554]: searchedCVE should not be mandatory. And if it is not present, all CVEs should be returned, just like current behavior on main.
```suggestion
        searchedCVE: String
```

File: pkg/extensions/search/cve/cve.go [hunk 2]
```diff
@@ -215,7 +216,18 @@ func (cveinfo BaseCveInfo) GetImageListWithCVEFixed(repo, cveID string) ([]commo
 	return fixedTags, nil
 }
 
-func (cveinfo BaseCveInfo) GetCVEListForImage(repo, tag string, pageInput PageInput) (
+func filterCVEList(cveMap map[string]cvemodel.CVE, searchedCVE string, pageFinder *CvePageFinder) {
+	searchedCVE = strings.ToUpper(searchedCVE)
+
```
Comment (andaaron) [inline pkg/extensions/search/cve/cve.go:221]: So what happens if searchedCVE is empty string?

File: pkg/cli/cve_cmd.go [hunk 3]
```diff
@@ -123,6 +126,7 @@ func setupCveFlags(cveCmd *cobra.Command, variables cveFlagVariables) {
 		`zot server in USERNAME:PASSWORD format`)
 	cveCmd.Flags().StringVarP(variables.outputFormat, "output", "o", "", "Specify output format [text/json/yaml]."+
 		" JSON and YAML format return all info for CVEs")
+	cveCmd.Flags().StringVarP(variables.searchedCVE, "search", "s", "", "Search specific CVEs by name/id")
```
Comment (andaaron) [inline pkg/cli/cve_cmd.go:129]: I think this should be saved together with the image name and cveID in the map searchCveParams. Is that possible?
The code would be cleaner.

```suggestion
	variables.searchCveParams["searchedCVE"] = cveCmd.Flags().StringP("search", "s", "", Search specific CVEs by name/id")
```

File: pkg/extensions/search/cve/cve.go (hunk 2 above)
Comment (aokirisaki) [reply pkg/extensions/search/cve/cve.go:221]: It will match all the CVEs, because all CVE titles/ids will contain the "" string

File: pkg/extensions/search/cve/cve.go [hunk 4]
```diff
@@ -19,7 +20,7 @@ import (
 type CveInfo interface {
 	GetImageListForCVE(repo, cveID string) ([]common.TagInfo, error)
 	GetImageListWithCVEFixed(repo, cveID string) ([]common.TagInfo, error)
-	GetCVEListForImage(repo, tag string, pageinput PageInput) ([]cvemodel.CVE, PageInfo, error)
+	GetCVEListForImage(repo, tag string, pageinput PageInput, searchedCVE *string) ([]cvemodel.CVE, PageInfo, error)
```
Comment (andaaron) [inline pkg/extensions/search/cve/cve.go:23]: I think we should convert it from pointer to string before calling this function. For consistency with the parameters `repo` and `tag`. Also reorder the parameters for `pageinput` to be last.

File: pkg/cli/cve_cmd.go [hunk 5]
```diff
@@ -83,9 +83,10 @@ func NewCveCommand(searchService SearchService) *cobra.Command {
 				fixedFlag:     &fixedFlag,
 				verifyTLS:     &verifyTLS,
 				verbose:       &verbose,
-				debug:         &debug,
-				resultWriter:  cmd.OutOrStdout(),
-				spinner:       spinnerState{spin, isSpinner},
+
```
Comment (andaaron) [inline pkg/cli/cve_cmd.go:86]: Revert this change, is it not needed.

**Review by andaaron — APPROVED** [review]
Looks good.

**Review by rchincha — APPROVED** [review]
lgtm
